token/js/test/e2e-2022/immutableOwner.test.ts

```diff
@@ -0,0 +1,69 @@
+import chai, { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+
+import {
+    Connection,
+    Keypair,
+    PublicKey,
+    Signer,
+    SystemProgram,
+    Transaction,
+    sendAndConfirmTransaction,
+} from '@solana/web3.js';
+
+import {
+    AuthorityType,
+    setAuthority,
+    ExtensionType,
+    createInitializeImmutableOwnerInstruction,
+    getAccountLen,
+} from '../../src';
+
+import { TEST_PROGRAM_ID, newAccountWithLamports, getConnection } from '../common';
+
+const EXTENSIONS = [ExtensionType.ImmutableOwner];
+describe('immutableOwner', () => {
+    let connection: Connection;
+    let payer: Signer;
+    let owner: Keypair;
+    let account: PublicKey;
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+    });
+    beforeEach(async () => {
+        owner = Keypair.generate();
+        const accountLen = getAccountLen(EXTENSIONS);
+        const lamports = await connection.getMinimumBalanceForRentExemption(accountLen);
+        const accountKeypair = Keypair.generate();
+        account = accountKeypair.publicKey;
+        const transaction = new Transaction().add(
+            SystemProgram.createAccount({
+                fromPubkey: payer.publicKey,
+                newAccountPubkey: account,
+                space: accountLen,
+                lamports,
+                programId: TEST_PROGRAM_ID,
+            }),
+            createInitializeImmutableOwnerInstruction(account, TEST_PROGRAM_ID)
+        );
```

Comment:
In order to properly create an immutable account, you'll need to do two transactions: one to create the mint, and a second to create the account. You can create the mint using:
```
          const mintAuthority = Keypair.generate();
          const mintKeypair = Keypair.generate();
          mint = await createMint(
              connection,
              payer,
              mintAuthority.publicKey,
              mintAuthority.publicKey,
              TEST_TOKEN_DECIMALS,
              mintKeypair,
              undefined,
              TEST_PROGRAM_ID
          );
```
and then create the account by doing:
```
        const transaction = new Transaction().add(
            SystemProgram.createAccount({
                fromPubkey: payer.publicKey,
                newAccountPubkey: account,
                space: accountLen,
                lamports,
                programId: TEST_PROGRAM_ID,
            }),
            createInitializeImmutableOwnerInstruction(account, TEST_PROGRAM_ID),
            createInitializeAccountInstruction(account, mint, owner, TEST_PROGRAM_ID),
        );
        await sendAndConfirmTransaction(connection, transaction, [payer, accountKeypair], undefined);
```
Note: this was pretty much copied from your code and the setAuthority tests